### PR TITLE
Add STL to Time Series Documentation

### PR DESCRIPTION
**Pull request by eccabay**
Adds a section about STL decomposition to the docs. 

Also, removed the `Could not find requested time_index None` warning that was popping up unnecessarily, because it was annoying.

### CODE REVIEW

File: docs/source/user_guide/timeseries.ipynb [hunk 1]
```diff
@@ -314,18 +321,7 @@
...
-    "axs[0].plot(y, \"r\")\n",
-    "axs[0].set_title(\"signal\")\n",
-    "axs[1].plot(res[0][\"trend\"], \"b\")\n",
-    "axs[1].set_title(\"trend\")\n",
-    "axs[2].plot(res[0][\"seasonality\"], \"g\")\n",
-    "axs[2].set_title(\"seasonality\")\n",
-    "axs[3].plot(res[0][\"residual\"], \"y\")\n",
-    "axs[3].set_title(\"residual\")\n",
-    "plt.show()"
+    "res = pdc.plot_decomposition(X_train_time, y_train_time)"
```
Comment (jeremyliweishih) [inline docs/source/user_guide/timeseries.ipynb:324]: the image isn't showing [here](https://feature-labs-inc-evalml--3835.com.readthedocs.build/en/3835/user_guide/timeseries.html#Polynomial-Decomposer) for some reason - can you double check?

**Review by jeremyliweishih — APPROVED** [review]
LGTM - just one thing of note

File: docs/source/user_guide/timeseries.ipynb (hunk 1 above)
Comment (eccabay) [reply docs/source/user_guide/timeseries.ipynb:324]: I'm seeing it with no problem - are you sure it's missing?

Comment (jeremyliweishih) [reply docs/source/user_guide/timeseries.ipynb:324]: I can see it now - might have just been my browser!

**Review by chukarsten — APPROVED** [review]
Thanks for doing this!

**Pull request merged**